### PR TITLE
WIP - Add dependency checker module for reducing amount of native mode tests

### PR DESCRIPTION
**Pull request by aguibert**

### CODE REVIEW

File: .github/workflows/native-cron-build.yml [hunk 1]
```diff
@@ -29,8 +32,8 @@ jobs:
       - name: Checkout Quarkus
         uses: actions/checkout@v2
         with:
-          repository: quarkusio/quarkus
-          ref: master
+          repository: aguibert/quarkus
```
Comment (machi1990) [inline .github/workflows/native-cron-build.yml]: We should probably revert all changes in this file?

File: .github/workflows/native-cron-build.yml [hunk 2]
```diff
@@ -39,34 +42,21 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-maven-
 
+      - name: Run dependency checker
```
Comment (machi1990) [inline .github/workflows/native-cron-build.yml]: this should be in https://github.com/quarkusio/quarkus/blob/master/.github/workflows/ci-actions.yml?

File: independent-projects/utilities/dependency-checker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		<!-- <dependency> -->
+		<!-- <groupId>org.eclipse.jgit</groupId> -->
+		<!-- <artifactId>org.eclipse.jgit</artifactId> -->
+		<!-- <version>5.6.0.201912101111-r</version> -->
+		<!-- </dependency> -->
+		<!-- <dependency> -->
+		<!-- <groupId>org.slf4j</groupId> -->
+		<!-- <artifactId>slf4j-log4j12</artifactId> -->
+		<!-- <version>1.7.26</version> -->
+		<!-- </dependency> -->
```
Comment (machi1990) [inline independent-projects/utilities/dependency-checker/pom.xml]: Can be removed?

File: .github/workflows/native-cron-build.yml [hunk 4]
```diff
@@ -39,34 +42,21 @@ jobs:
...
+        run: |
+          echo "@AGG Inside of dependency checker task"
+          echo "GITHUB_SHA=${GITHUB_SHA}"
+          echo "GITHUB_REF=$GITHUB_REF"
+          echo "GITHUB_HEAD_REF=$GITHUB_HEAD_REF"
+          echo "GITHUB_BASE_REF=$GITHUB_BASE_REF"
+          mvn -B -f independent-projects/utilities/dependency-checker/pom.xml package
+          java -jar independent-projects/utilities/dependency-checker/target/quarkus-utilities-dependency-checker-999-SNAPSHOT-jar-with-dependencies.jar ${GITHUB_BASE_REF} ${GITHUB_HEAD_REF}
```
Comment (aguibert) [inline .github/workflows/native-cron-build.yml]: This section is just a WIP while I figure out how to properly hook the tool into actions

File: .github/workflows/native-cron-build.yml (hunk 1 above)
Comment (aguibert) [reply .github/workflows/native-cron-build.yml]: hi @machi1990, thanks for having a look at the PR. All the workflow changes right will be rolled back. I've hacked up the workflow files just to make the testing quicker

Comment (machi1990) [reply .github/workflows/native-cron-build.yml]: Thanks, realised that I jumped quickly with the review and missed your comment about this being a WIP.

**Comment by aguibert** [comment]
For #6260

**Comment by gsmet** [comment]
@famod isnt what you're working on supposed to replace this?

**Comment by famod** [comment]
@gsmet Since I wasn't really aware of this until now, I was/am not aiming to "replace" this but yeah this is more or less the same approach.
My approach with GIB would leave all those fine grained module-specific jobs untouched, though. You always have to be prepared for a full build and one single job/build might blow up then (timeouts etc.).

**Comment by gastaldi** [comment]
Closing as this needs a rebase and we're aiming to use GIB instead

**Pull request closed without merging**